### PR TITLE
feat(pt_expt): enhance per-module export tests + fix nframes symbol collision

### PR DESCRIPTION
**Pull request by wanghan-iapcm**
## Summary
- Enhance existing `test_make_fx` (9 descriptors, 6 fittings) and `test_forward_lower_exportable` (8 models) to cover the full `dp freeze` pipeline: symbolic tracing, `torch.export` with dynamic shapes, `.pte` save/load round-trip, and inference with different `nframes`
- Add `**make_fx_kwargs` forwarding to 6 model `forward_lower_exportable` methods (dipole, dos, polar, property, dp_zbl, dp_linear)
- Create `source/tests/pt_expt/export_helpers.py` with shared test helpers
- Add new `test_make_fx` to `test_ener_fitting.py` and `test_invar_fitting.py` with fparam/aparam parametrization
- Fix nframes symbol collision: change tracing `nframes` from 2 to 5 in `_trace_and_export` to avoid PyTorch's symbolic tracer merging dim symbols when `nframes == numb_fparam`

## Test plan
- [x] 16 descriptor `test_make_fx` pass
- [x] 6 fitting `test_make_fx` pass
- [x] 8 model `test_forward_lower_exportable` pass
- [x] `test_export_pipeline` with `fparam=True` passes for se_e2_a and dpa1
- [ ] CI

## Summary by CodeRabbit

* **New Features**
  * FX tracing for export is now caller-configurable for exportable model methods.
  * Increased sample frames used for export traces to avoid unwanted specialization/collisions.

* **Tests**
  * Added comprehensive export → save → load round-trip validations across descriptor, fitting, and model tests.
  * Introduced shared test helpers/utilities for dynamic-shape export verification.
  * Added export-pipeline and linear-model test coverage; many tests refactored to use the new helpers and one obsolete test module removed.

### CODE REVIEW

File: source/tests/pt_expt/model/test_export_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+        # 7. .pte save → load round-trip
+        with tempfile.NamedTemporaryFile(suffix=".pte", delete=True) as tmp:
+            tmpfile = tmp.name
+        torch.export.save(exported, tmpfile)
+        loaded = torch.export.load(tmpfile).module()
+
+        # Clean up temp file
+        import os
+
+        os.unlink(tmpfile)
```
Comment (iProzd) [inline source/tests/pt_expt/model/test_export_pipeline.py]: Race condition in temp file handling. The `NamedTemporaryFile(delete=True)` deletes the file when the with block exits, then torch.export.save writes to the now-deleted path. Another process could claim that path in between. 
This also differs from the pattern used everywhere else in this PR:
```
with tempfile.NamedTemporaryFile(suffix=".pte") as f:
    torch.export.save(exported, f.name)
    loaded = torch.export.load(f.name).module()
```
Suggest using the same with-block pattern for consistency and safety, and dropping the manual `os.unlink`.

File: source/tests/pt_expt/descriptor/test_dpa1.py
Comment (iProzd) [inline source/tests/pt_expt/descriptor/test_dpa1.py:1]: The 34-line export round-trip block (symbolic trace → compare → nf=1 slice → compare) is copy-pasted nearly identically across 9 descriptor tests and 6 fitting tests. Since `export_helpers.py` already exists, consider adding a helper that also covers the "different nframes" verification, so each call site shrinks to ~5 lines. Same applies to the ~45-line block repeated across model tests (dipole/dos/polar/property/dp_zbl).

File: source/tests/pt_expt/model/test_linear_model.py
Comment (iProzd) [inline source/tests/pt_expt/model/test_linear_model.py:1]: There's already a `test_linear_ener_model.py` in the same directory. What's the relationship between them?

**Review by iProzd — CHANGES_REQUESTED** [review]

File: source/tests/pt_expt/model/test_export_pipeline.py (hunk 1 above)
Comment (wanghan-iapcm) [reply source/tests/pt_expt/model/test_export_pipeline.py]: resolved by [f68a659](https://github.com/deepmodeling/deepmd-kit/pull/5367/commits/f68a659256aad37ab0003a5e2eed81748da81e8a)

File: source/tests/pt_expt/descriptor/test_dpa1.py
Comment (wanghan-iapcm) [reply source/tests/pt_expt/descriptor/test_dpa1.py:1]: resolved by [f68a659](https://github.com/deepmodeling/deepmd-kit/pull/5367/commits/f68a659256aad37ab0003a5e2eed81748da81e8a)

File: source/tests/pt_expt/model/test_linear_model.py
Comment (wanghan-iapcm) [reply source/tests/pt_expt/model/test_linear_model.py:1]: merged the two files.

File: source/tests/pt_expt/model/test_export_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+                ext_coord, ext_atype, nlist_t, mapping_t, fparam_ones, aparam
+            )
+            # Output with fparam=0 should differ from fparam=1
+            assert not np.allclose(
+                loaded_out["energy"].detach().cpu().numpy(),
+                loaded_out_fp1["energy"].detach().cpu().numpy(),
+            ), (
+                "Changing fparam did not change output — "
+                "fparam may be baked in as a constant"
+            )
```
[the remark this answers is not in the document]
Comment (wanghan-iapcm) [reply source/tests/pt_expt/model/test_export_pipeline.py:265]: fixed by 03953dd3645eee76875db08a9cc69efd6288d3ac

**Review by iProzd — APPROVED** [review]
